File: AutomaticDrawing.Utilities.Bitmap/Bitmap/Extentions/BitmapExtensions.vb
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports AutomaticDrawing.Core

Public Module BitmapExtensions
    ''' <summary>
    ''' 返回当前位图的像素数据
    ''' </summary>
    <Extension>
    Public Function GetPixelData(this As Drawing.Bitmap) As PixelData
        Return BitmapHelper.GetPixelDataFromBitmap(this)
    End Function
    ''' <summary>
    ''' 从当前像素数据生成位图
    ''' </summary>
    <Extension>
    Public Function CreateBitmap(this As PixelData) As Drawing.Bitmap
        Return BitmapHelper.GetBitmapFromPixelData(this)
    End Function
End Module

File: AutoPaint.Utilities.Bitmap/Bitmap/Extensions/BitmapExtensions.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using AutoPaint.Core;

namespace AutoPaint.Utilities
{
    public static class BitmapExtensions
    {
        public static PixelData GetPixelData(this Bitmap bitmap)
        {
            return BitmapHelper.GetPixelDataFromBitmap(bitmap);
        }

        public static Bitmap CreateBitmap(this PixelData bitmap)
        {
            return BitmapHelper.GetBitmapFromPixelData(bitmap);
        }
    }

}
